daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/BinaryNumberTraits.scala

```diff
@@ -103,6 +103,18 @@ trait PrefixedLengthParserMixin {
         "Prefixed length result must be non-negative after dfdl:prefixIncludesPrefixLength adjustment , but was: %d",
         adjustedLen
       )
+      // do checks on facets expressed on prefixLengthType
+      val optSTRD = plElement.erd.optSimpleTypeRuntimeData
+      if (optSTRD.isDefined) {
+        val strd = optSTRD.get
+        val check = strd.executeCheck(plElement)
+        if (check.isError) {
+          val pe = state.toProcessingError(
+            s"The calculated value of ${prefixedLengthERD.namedQName} ($adjustedLen) failed check due to ${check.errMsg}"
```

Comment:
I wonder if this message should include `parsedLen` instead of `adjustedLen`?

For example, say we have `prefixIncludesPrefixLength="yes"` in the `PrefixFacetsCheck` test below. In that case the adjusted length would be 4 (since we need to subtract the length of the 1-byte prefix element). The facet check would still fail since it's checked against the parsedLength in the infoset (5), but we would output an error like"

>  s"The calculated value of prefix2 (4) failed check due to facet maxInclusive (4)"

Which would be confusing since that looks fine.

That said, maybe the facet checks should be run on the adjusted value and not the parsed value? In which case we would need to change the infoset value to the adjusted length prior to calling executeCheck. I think the spec implies facet checks are run on the parsed value since there's no mention of adjustment.

@mbeckerle, can you clarify?

Reply:
Hmmm. DFDL Workgroup discussion did not catch this level of detail that I recall. I'll search around a bit or ask and report back.

Reply:
Note, due to some issues with the DFDL-WG email lists, this inquiry to the group did not get sent until morning of 2024-10-14.

Reply:
@mbeckerle , any word from the WG on this?

Reply:
> @mbeckerle , any word from the WG on this?

No, I've not seen any comments on my email on this subject. 

I think we should just proceed with the decision that the facets are about the value that is stored in the prefix integer, not the length of the data. So if prefixIncludesPrefixLength is 'true' the value stored is the length of the data plus the length of the prefix, and that sum is what the facets are constraining. 

So suppose the data is a string with a 2 byte unsigned prefix and prefixIncludesPrefixLength is true and the max string length is 1000. Then I believe these things hold:

- when parsing: a prefix value of 0 or 1 is malformed and causes a parse error
- when parsing: a prefix value of 2 means zero-length
- The maxInclusive facet should be 1002, as that means a string  of maximum length 1000. 
- a prefix value of 1003 or above is malformed and causes a parse error.